rama-pac/src/pac_parser/mod.rs

```diff
@@ -0,0 +1,90 @@
+use boa_engine::{Context as JsContext, JsString, JsValue, Source, js_string, property::Attribute};
+use boa_runtime::Console;
+use rama_core::error::BoxError;
+use rama_http::Uri;
+
+const PAC_STDLIB: &str = r#"
+// spec-ish helpers
+function dnsDomainIs(host, domain) {
```

Comment:
Instead of using these js interpreted functions, use rust host functions please, so we can use std rama code and it will be easier to keep it straight